snap/snapcraft.yaml

```diff
@@ -0,0 +1,107 @@
+name: bpftrace
+summary: high-level tracing language for Linux eBPF
+description: BPFtrace is a high-level tracing language for
+  Linux enhanced Berkeley Packet Filter available in recent
+  Linux kernels. BPFtrace uses LLVM as a backend to compile scripts
+  to BPF-bytecode and makes use of BCC for interacting with the Linux
+  BPF system, as well as existing Linux tracing capabilities like
+  kernel dynamic tracing, user-level dynamic tracing and tracepoints.
+
+confinement: strict
+grade: stable
+base: core18
+architectures: [amd64]
+assumes: [snapd2.37]
+adopt-info: bpftrace
+
+parts:
+    bcclibs3:
+        source: https://github.com/iovisor/bcc.git
+        source-tag: v0.8.0
+        plugin: cmake
+        configflags:
+            - -DLLVM_DEFINITIONS="-D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS"
+            - -DREVISION_LAST=v0.8.0
+            - -DBCC_KERNEL_HAS_SOURCE_DIR=1
+            - -DPYTHON_CMD='python2;python3'
+            - -DCMAKE_BUILD_PARALLEL_LEVEL=1
+            - -DCMAKE_INSTALL_PREFIX=/usr
+        override-pull: |
+          snapcraftctl pull
+          sed -i "s;if os.environ.get('DESTDIR'):;if \"sdist\" not in sys.argv and os.environ.get('DESTDIR'):;" src/python/setup.py.in
+        build-packages:
+            - git
+            - libc6
+            - iperf
+            - bison
+            - cmake
+            - flex
+            - g++
+            - binutils-dev
+            - libstdc++-dev
+            - libelf-dev
+            - zlib1g-dev
+            - libfl-dev
+            - clang-6.0
+            - libclang-6.0-dev
+            - libclang-common-6.0-dev
+            - libclang1-6.0
+            - libllvm6.0
+            - llvm-6.0
+            - llvm-6.0-dev
+            - llvm-6.0-runtime
+            - systemtap-sdt-dev
+            - netperf
+            - python3
+            - arping
+            - clang-format
+            - ethtool
+            - inetutils-ping
+            - libedit-dev
+            - libzip-dev
+            - libluajit-5.1-dev
+            - luajit
+            - python3-netaddr
+            - python3-pyroute2
+            - python3-distutils
+    bpftrace:
+        after:
+            - bcclibs3
+        source: https://github.com/iovisor/bpftrace.git
+        plugin: cmake
+        configflags:
+            - -DCMAKE_BUILD_TYPE=Release
+            - -DLIBBCC_INCLUDE_DIRS=$SNAPCRAFT_STAGE/usr/include/bcc
+        override-pull: |
+            snapcraftctl pull
+            version=$(git tag | sort -V | tail -1 | cut -c2-)
+            commits=$(git log --oneline | wc -l)
+            sha=$(git log -1 --oneline | cut -d' ' -f1)
```

Comment:
we use `git describe` to determine the version number for our git-builds, it does most of this for you:

```
$ git describe head --tags
v0.9.4-83-g11814f2
```

Reply:
I've updated the commit to use git describe HEAD --tags